Detectors/ITSMFT/ITS/RawPixelReaderWorkflow/src/RawPixelReaderSpec.cxx

```diff
@@ -0,0 +1,608 @@
+        pc.outputs().snapshot(Output{ "TST", "Error", 0, Lifetime::Timeframe }, ErrorVecTest[0]);
+        pc.outputs().snapshot(Output{ "TST", "Finish", 0, Lifetime::Timeframe }, FileInfo);
+        pc.outputs().snapshot(Output{ "ITS", "DIGITS", 0, Lifetime::Timeframe }, mMultiDigitsTest);
+        NewFileInj = 0;
+        ErrorVecTest.clear();
+        mDigitsTest.clear();
+        mMultiDigitsTest.clear();
+        if (FolderNames.size() < NowFolderNames.size())
+          FileNames.push_back(NewNextFold);
+        cout << "Done!!! You should see the Canvas Updated " << endl;
+        break;
+      }
+
+      //DONE Inhection//
+
+      o2::itsmft::RawPixelReader<o2::itsmft::ChipMappingITS> rawReader;
+      rawReader.setPadding128(true); // payload GBT words are padded to 16B
+      rawReader.setVerbosity(0);
+      rawReader.setMinTriggersToCache(1025);
+
+      rawReader.openInput(inpName);
+      //mDigits.clear();
+      //mMultiDigits.clear();
+
+      int Index = 0;
+      int IndexMax = -1;
+      int NChip = 0;
+      int NChipMax = -1;
+      int TimePrint = 0;
+      using RawReader = o2::itsmft::RawPixelReader<o2::itsmft::ChipMappingITS>;
+      auto& rawErrorReader = reinterpret_cast<RawReader&>(rawReader);
+
+      startDecoder = std::chrono::high_resolution_clock::now();
+
+      while (mChipData = rawReader.getNextChipData(mChips)) {
+        if (NChip < NChipMax)
+          break;
+        //	cout << "Pass Chip" << endl;
+
+        const auto* ruInfo = rawErrorReader.getCurrRUDecodeData()->ruInfo;
+        const auto& statRU = rawErrorReader.getRUDecodingStatSW(ruInfo->idSW);
+
+        const auto& pixels = mChipData->getData();
+        int PixelSize = mChipData->getData().size();
+
+        NEvent = statRU->nPackets;
+
+        TotalPixelSize = TotalPixelSize + PixelSize;
+
+        if (NEvent > (EventRegistered + 1) * EventPerPush) {
+
+          if (TotalPixelSize < MaxPixelSize || TotalPixelSize == MaxPixelSize) {
+            cout << "Digit OK for 1 Push" << endl;
+            NDigits.push_back(TotalPixelSize);
+            EventRegistered = EventRegistered + 1;
+            ErrorVec.push_back(Error);
+            cout << "TotalPixelSize = " << TotalPixelSize << "  Pushed" << endl;
+          }
+
+          if (TotalPixelSize > MaxPixelSize) {
+            cout << "Digit Spilt into 2 Pusbhes" << endl;
+            NDigits.push_back(TotalPixelSize / 2);
+            NDigits.push_back(TotalPixelSize / 2);
+            ErrorVec.push_back(Error);
+            ErrorVec.push_back(Error);
+            EventRegistered = EventRegistered + 1;
+            cout << "TotalPixelSize = " << TotalPixelSize << "  Pushed" << endl;
+          }
+
+          TotalPixelSize = 0;
+        }
+
+        if (NEvent % 1000000 == 0 && TimePrint == 0) {
+          cout << "Event Number = " << NEvent << endl;
+          TimePrint = 1;
+        }
+
+        if (NEvent % 100000 != 0)
+          TimePrint = 0;
+
+        if (Error[0] < 4294967295)
```

Comment:
Why don't you simply loop:

````
for (int i=0;i<o2::itsmft::GBTLinkDecodingStat::NErrorsDefined; i++) {
   Error[i] += (int)statRU->errorCounts[i];
}
````
Note that in future these error types will change for sure, better avoid to chaing of changes in dependent code. In fact, it would be better to use directly ``o2::itsmft::GBTLinkDecodingStat`` struct instance instead of the Error array. 
You can simply equip it with ``operator+()`` and do 
````
o2::itsmft::GBTLinkDecodingStat mErrors;
...
 mErrors += *statRU;
````

Reply:
Suggestion 1 has been taken!